pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1839,173 +1257,112 @@ func TestServiceRegistryList(t *testing.T) {
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	createdSvc1 := obj.(*api.Service)
+	if createdSvc1.Name != "foo" {
+		t.Errorf("Expected foo, but got %v", createdSvc1.Name)
+	}
+	if !makeIPNet(t).Contains(net.ParseIP(createdSvc1.Spec.ClusterIPs[0])) {
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc1.Spec.ClusterIPs[0])
+	}
+
+	svc2 := svctest.MakeService("bar")
 	ctx = genericapirequest.NewDefaultContext()
-	createdSvc2, _ := storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
-	createdService2 := createdSvc2.(*api.Service)
-	if createdService2.Name != "bar" {
-		t.Errorf("Expected bar, but got %v", createdService2.Name)
+	obj, err = storage.Create(ctx, svc2, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error creating service: %v", err)
+	}
+	createdSvc2 := obj.(*api.Service)
+	if createdSvc2.Name != "bar" {
+		t.Errorf("Expected bar, but got %v", createdSvc2.Name)
 	}
-	if !makeIPNet(t).Contains(net.ParseIP(createdService2.Spec.ClusterIPs[0])) {
-		t.Errorf("Unexpected ClusterIP: %s", createdService2.Spec.ClusterIPs[0])
+	if !makeIPNet(t).Contains(net.ParseIP(createdSvc2.Spec.ClusterIPs[0])) {
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc2.Spec.ClusterIPs[0])
 	}
 
 	testIPs := []string{"1.2.3.93", "1.2.3.94", "1.2.3.95", "1.2.3.96"}
-	testIP := ""
+	testIP := "not-an-ip"
 	for _, ip := range testIPs {
 		if !storage.serviceIPAllocatorsByFamily[storage.defaultServiceIPFamily].(*ipallocator.Range).Has(net.ParseIP(ip)) {
 			testIP = ip
 			break
 		}
 	}
 
-	svc3 := &api.Service{
-		ObjectMeta: metav1.ObjectMeta{Name: "quux"},
-		Spec: api.ServiceSpec{
-			Selector:        map[string]string{"bar": "baz"},
-			ClusterIP:       testIP,
-			ClusterIPs:      []string{testIP},
-			SessionAffinity: api.ServiceAffinityNone,
-			Type:            api.ServiceTypeClusterIP,
-			Ports: []api.ServicePort{{
-				Port:       6502,
-				Protocol:   api.ProtocolTCP,
-				TargetPort: intstr.FromInt(6502),
-			}},
-		},
-	}
+	svc3 := svctest.MakeService("qux", svctest.SetClusterIPs(testIP))
 	ctx = genericapirequest.NewDefaultContext()
-	createdSvc3, err := storage.Create(ctx, svc3, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
+	obj, err = storage.Create(ctx, svc3, rest.ValidateAllObjectFunc, &metav1.CreateOptions{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	createdService3 := createdSvc3.(*api.Service)
-	if createdService3.Spec.ClusterIPs[0] != testIP { // specific IP
-		t.Errorf("Unexpected ClusterIP: %s", createdService3.Spec.ClusterIPs[0])
+	createdSvc3 := obj.(*api.Service)
+	if createdSvc3.Spec.ClusterIPs[0] != testIP { // specific IP
+		t.Errorf("Unexpected ClusterIP: %s", createdSvc3.Spec.ClusterIPs[0])
 	}
 }
 
 func TestServiceRegistryIPReallocation(t *testing.T) {
-	storage, _, server := NewTestREST(t, nil, singleStackIPv4)
+	storage, server := NewTestREST(t, singleStackIPv4)
```

Comment:
I think we can just use a static IP and cut all the code that tests in IPNet. Will greatly simplify the test.